nsv13/code/modules/overmap/ai.dm

```diff
@@ -24,34 +33,78 @@
 		pilot.forceMove(src)
 		gunner = pilot
 	user_thrust_dir = 1
-	if(last_target) //Have we got a target?
+	if(last_target || QDELETED(last_target)) //Have we got a target?
 		if(get_dist(last_target, src) > max_range) //Out of range - Give up the chase
 			last_target = null
 			desired_angle = rand(0,360)
 		else //They're in our range. Calculate a path to them and fire.
 			desired_angle = Get_Angle(src, last_target)
 			fire(last_target) //Fire already handles things like being out of range, so we're good
-	else
-		for(var/X in GLOB.overmap_objects)
-			if(!istype(X, /obj/structure/overmap))
-				continue
-			var/obj/structure/overmap/ship = X
-			if(ship == src || ship.faction == faction || ship.z != z)
-				continue
-			switch(ai_behaviour)
-				if(AI_AGGRESSIVE)
-					if(get_dist(ship,src) <= max_range)
-						target(ship)
-				if(AI_GUARD)
-					if(get_dist(ship,src) <= guard_range)
-						target(ship)
-				if(AI_RETALIATE)
-					if(ship in enemies)
-						target(ship)
+	handle_ai_behaviour()
+
+/obj/structure/overmap/proc/handle_ai_behaviour()
+	for(var/obj/structure/overmap/ship in GLOB.overmap_objects)
+		if(ship == src || ship.faction == faction || ship.z != z)
+			continue
+		ai_target(ship)
+
+
+/**
+*
+*
+* ai_target -> what happens after a targetable ship is found
+* if your ship subtype is "timid" like carriers, override this to handle ai behaviours specifically, otherwise, ship will be targeted based on its "ai behaviour"
+* @param ship overmap ship that was found by handle_ai_behaviour()
+*
+*
+*/
+
+/obj/structure/overmap/proc/ai_target(obj/structure/overmap/ship)
+	switch(ai_behaviour)
+		if(AI_AGGRESSIVE)
+			if(get_dist(ship,src) <= max_range)
+				target(ship)
+		if(AI_GUARD)
+			if(get_dist(ship,src) <= guard_range)
+				target(ship)
+		if(AI_RETALIATE)
+			if(ship in enemies)
+				target(ship)
+	if(locate(ship) in enemies)
+		if(get_dist(ship, src) <= 3)
+			user_thrust_dir = 0 //Don't thrust towards ships we're already close to.
+			brakes = TRUE
+		else
+			user_thrust_dir = 1
+			brakes = FALSE
+
+/**
+*
+* Carriers are more timid, and prefer to run away from enemy ships before firing, engaging at extreme range.
+*
+*/
+
+/obj/structure/overmap/syndicate/ai/carrier/ai_target(obj/structure/overmap/ship)
+	if(get_dist(ship,src) <= max_range)
+		if(get_dist(ship, src) <= max_range/2) //Little bit too friendly there partner.
+			last_target = ship
+			retreat()
+		else
+			target(ship)
+			try_launch_fighters()
+
+/obj/structure/overmap/proc/retreat()
+	if(!last_target)
+		return
+	desired_angle = -Get_Angle(src, last_target) //Turn the opposite direction and run.
+	user_thrust_dir = 1
```

Comment:
is this var always a 0 or 1 ? If so please do a TRUE and FALSE across the files.